src/components/manifold-plan-table/manifold-plan-table.tsx

```diff
@@ -48,20 +48,35 @@ interface PlanFeatures {
 
 // state
 type UserValue = string | number | boolean;
+type FeatureMap = { [featureLabel: string]: UserValue };
 interface UserSelection {
-  [planID: string]: { [featureLabel: string]: UserValue };
+  [planID: string]: FeatureMap;
+}
+
+// event
+
+interface PlanTableInit {
+  userSelection: any;
+}
+
+interface PlanTableEvent {
+  planID: string;
+  planDisplayName: string;
+  userSelection: FeatureMap;
 }
 
 @Component({ tag: 'manifold-plan-table' })
 export class ManifoldPlanTable {
   @Element() el: HTMLElement;
-  @Event() CTAClick: EventEmitter;
+  @Event({ bubbles: false }) ctaClick: EventEmitter<PlanTableEvent>;
+  @Event({ bubbles: false }) init: EventEmitter<PlanTableInit>;
+  @Event({ bubbles: false }) update: EventEmitter<PlanTableEvent>;
```

Comment:
I don’t know why we didn’t use these generics for event types in UI 💯

Reply:
We did in a couple places! Check out `manifold-oauth` for example.